dvc/tree/local.py

```diff
@@ -325,7 +331,7 @@ def _upload(
             from_file, tmp_file, name=name, no_progress_bar=no_progress_bar
         )
 
-        self.protect(tmp_file)
+        self.protect(tmp_file, mode=file_mode)
```

Comment:
The `protect` naming no longer makes sense if it can accept just arbitrary modes :slightly_frowning_face: But I guess it didn't make sense before, this should really be something like `self.chmod(tmp_file, file_mode)` and we should somehow handle `protect` logic in a more logical way.